### PR TITLE
Feature redux actions

### PR DESCRIPTION
**Pull request by AndresMWeber**

### CODE REVIEW

**Review by bobbypwang — APPROVED** [review]

File: server/routes/games.js [hunk 1]
```diff
@@ -1,23 +1,131 @@
...
+ * @example
+ * GET /api/games/search?limit=50
+ * */
+router.get('/search', async(req, res, next) => {
+    const limit = Number(req.query.limit) || 50
+    const terms = req.query.terms || ''
+
+    Game.find({ $text: { $search: terms } })
+        .limit(limit)
+        .populate('likes')
```
Comment (Kiwi-x-Kiwi) [inline server/routes/games.js:38]: Should also populate categories

File: server/routes/games.js [hunk 2]
```diff
@@ -1,23 +1,131 @@
...
+        .catch(err => next(err))
+})
+
+/** 
+ * Get all games.  
+ * @example
+ * GET /api/games/
+ * */
 router.get('/', async(req, res, next) => {
+    res.json(await Game.find().populate('likes'))
```
Comment (Kiwi-x-Kiwi) [inline server/routes/games.js:51]: Also populate categories here

File: server/routes/games.js [hunk 3]
```diff
@@ -1,23 +1,131 @@
...
+ */
+router.get('/:id', async(req, res, next) => {
     try {
-        const response = await axios
-            .get("https://www.boardgameatlas.com/api/search", {
-                params: {
-                    client_id: process.env.BOARD_GAME_ATLAS_CLIENT_ID,
-                    limit: 100
-                }
+        const game = await Game.findById(req.params.id).populate('likes')
```
Comment (Kiwi-x-Kiwi) [inline server/routes/games.js:79]: populate categories here too

File: server/routes/games.js [hunk 4]
```diff
@@ -1,23 +1,131 @@
...
+        if (!game) throw new Error()
+        if (req.user.role !== "admin") res.status(403).send('You do not have permission to delete this resource.')
+        Like.deleteMany({ game: game._id })
+            .then(
+                await game.remove()
+            )
+            .catch(err => {
+                console.log(err)
+                res.status(404).send(err)
             })
```
Comment (Kiwi-x-Kiwi) [inline server/routes/games.js]: I added this feature to the Schema.  Will create a pull request later, we can remove this when that PR gets merged.

File: server/routes/games.js [hunk 5]
```diff
@@ -1,23 +1,131 @@
...
 
+/**
+ * Update a specific game
+ * @example 
+ * PATCH /api/games/:id
+ */
+router.patch(`/:id`, isLoggedIn, async(req, res) => {
+    if (req.user.role !== "admin") res.status(403).send('You do not have permission to update this resource.')
+
+    const updates = Object.keys(req.body).filter(e => fields.includes(e))
```
Comment (Kiwi-x-Kiwi) [inline server/routes/games.js]: Should we stick to what we did for the other routes where we send a error back when an invalid field is in `req.body`?

If not, should we keep consistency in the previous files and just update the valid field without returning an error?

**Review by Kiwi-x-Kiwi — CHANGES_REQUESTED** [review]
Need to populate `categories `in addition to `likes `in `server/routes/user.js`.  Left a few comments in that file too, if you can please read over them.

File: server/routes/games.js (hunk 1 above)
Comment (AndresMWeber) [reply server/routes/games.js:38]: It ain't a virtual!

File: server/routes/games.js (hunk 2 above)
Comment (AndresMWeber) [reply server/routes/games.js:51]: It ain't a virtual!

File: server/routes/games.js (hunk 3 above)
Comment (AndresMWeber) [reply server/routes/games.js:79]: It ain't a virtual!

File: server/routes/games.js (hunk 4 above)
Comment (AndresMWeber) [reply server/routes/games.js]: Removing!

File: server/routes/games.js (hunk 5 above)
Comment (AndresMWeber) [reply server/routes/games.js]: Yes!

**Pull request merged**